hw/hal/include/hal/hal_system.h

```diff
@@ -66,8 +66,10 @@ enum hal_reset_reason {
     HAL_RESET_SOFT = 4,
     /** Low supply voltage */
     HAL_RESET_BROWNOUT = 5,
+    /** Power on */
```

Comment:
I would change it to HAL_RESET_SYSTEM_OFF or such. Now we have comment stating Power on (which is double confusing:  first see comment on HAL_RESET_POR, second name is HAL_RESET_POWER_OFF)

Reply:
Updated comment to better reflect as to what is happening and to reduce confusion. 

`    /** System Off, wakeup on external interrupt*/`